lib/doh.c

```diff
@@ -969,6 +1313,20 @@ CURLcode Curl_doh_is_resolved(struct Curl_easy *data,
     } /* address processing done */
 
     /* Now process any build-specific attributes retrieved from DNS */
+#ifdef USE_HTTPSRR
+    if(de.numhttps_rrs > 0 && result == CURLE_OK && *dnsp) {
+      int rv;
+      struct Curl_https_rrinfo *hrr = NULL;
+      rv = Curl_doh_decode_httpsrr(de.https_rrs->val, de.https_rrs->len, &hrr);
+      if(rv != CURLE_OK) {
```

Comment:
We typically check for errors without comparing and just do it if `result` is non-zero:

```suggestion
      if(result) {
```

Reply:
done